src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -71,6 +71,7 @@ public sealed partial class TokenCache : ITokenCacheInternal
                         requestParams.AuthenticationScheme.KeyId)
                     {
                         UserAssertionHash = requestParams.UserAssertion?.AssertionHash,
+                        OboCacheKey = requestParams.OboCacheKey,
```

Comment:
Would it be possible to leave this class alone, i.e. just rename UserAssertionHash to OboSessionKey
Then the callers of this class would make sure to set `requestParams.OboSessionKey` to either user key or assertion hash?